geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/AcceptorImpl.java

```diff
@@ -1493,6 +1500,43 @@ protected void handleNewClientConnection(final Socket socket,
     }
   }
 
+  static final byte REPLY_REFUSED = (byte) 60;
+  static final byte REPLY_INVALID = (byte) 61;
+
+  void refuseHandshake(OutputStream out, String message, byte exception) throws IOException {
+
+    HeapDataOutputStream hdos = new HeapDataOutputStream(32, Version.CURRENT);
+    DataOutputStream dos = new DataOutputStream(hdos);
+    // Write refused reply
+    dos.writeByte(exception);
+
+    // write dummy endpointType
+    dos.writeByte(0);
+    // write dummy queueSize
+    dos.writeInt(0);
+
+    // Write the server's member
+    DistributedMember member = InternalDistributedSystem.getAnyInstance().getDistributedMember();
+    HeapDataOutputStream memberDos = new HeapDataOutputStream(Version.CURRENT);
```

Comment:
Now this code has migrated to another class but I think it still has the same problem.

Reply:
Let me know if there's a bug here.  The behavior shouldn't have changed, but I'm not entirely up to speed on the serialization code to see if there's an previous issue.